src/main/java/approximations/java/util/zip/CRC32_Tests.java

```diff
@@ -0,0 +1,183 @@
+package approximations.java.util.zip;
+
+import approximations.Test;
+
+import java.nio.ByteBuffer;
+import java.util.zip.CRC32;
+
+public final class CRC32_Tests {
+
+    // internal variables
+
+    // constructors
+
+    // CRC32Automaton::CRC32
+    public static int test_CRC32_0(final int execution) {
+        CRC32 crc32 = new CRC32();
+        if (execution == 0) {
+            long crc = crc32.getValue();
+            if (crc != 0) return -1;
+            else return execution;
+        }
+        return -1;
+    }
+
+
+    // static methods
+
+    // methods
+
+    // CRC32Automaton::getValue
+    public static int test_getValue_0(final int execution) {
+        CRC32 crc32 = new CRC32();
+        if (execution == 0) {
+            long crc = crc32.getValue();
+            if (crc != 0) return -1;
+            else return execution;
+        }
+        return -1;
+    }
+
+
+    // CRC32Automaton::reset
+    public static int test_reset_0(final int execution) {
+        CRC32 crc32 = new CRC32();
+        if (execution == 0) {
+            crc32.update(123456);
+            crc32.reset();
+            long crc = crc32.getValue();
+            if (crc != 0) return -1;
+            else return execution;
+        }
+        return -1;
+    }
+
+
+    // CRC32Automaton::update (ByteBuffer)
+    @Test(executionMax = 3)
```

Comment:
Buffers returned by `ByteBuffer#wrap` and `ByteBuffer#allocateDirect` heavily rely on using `jdk.internal.misc.Unsafe` which we have no approximations for yet making it impossible for the USVM to chew-through its `native` methods.

Consider `disable`ing this test for the time being (with a `reason`able comment).